source/extensions/filters/network/dubbo_proxy/router/router_impl.h

```diff
@@ -91,6 +93,7 @@ class Router : public Tcp::ConnectionPool::UpstreamCallbacks,
   void cleanup();
 
   Upstream::ClusterManager& cluster_manager_;
+  Envoy::Router::MetadataMatchCriteriaConstPtr metadata_match_;
```

Comment:
This needs to be initialized.

Reply:
This will be initialized when calling `Router::metadataMatchCriteria()`.

```
  if (route_criteria != nullptr) {
    metadata_match_ = route_criteria->mergeMatchCriteria(filter_it->second);
  } else {
    metadata_match_ = std::make_unique<Envoy::Router::MetadataMatchCriteriaImpl>(filter_it->second);
  }
```

Reply:
Alright, I don't have strong opinion on it. The type name suggests it's `std::unique_ptr` underneath, so should be safe to keep it as it is.

Reply:
ha, i get what you mean. 🤣 yeah, it's safe, the member will be initialized to `nullptr` by default.